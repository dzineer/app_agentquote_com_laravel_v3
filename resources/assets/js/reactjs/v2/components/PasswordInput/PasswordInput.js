import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ProgressBar from '../ProgressBar';
import EyeIcon from '../EyeIcon';
import TextInput from '../TextInput';

/** Password input with integrated label, quality tips, and show password toggle  */
class PasswordInput extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showPassword: false
        };

        this.css = {
            for: {
                container: {
                    marginBottom: '16px'
                },
                input: {
                    border: 'solid 1px red'
                },
                error: {
                    color: 'red'
                }
            }
        };
    }
    /** this a class property that is not part of the official JS spec yet. But Babel will transpile it. */
    toggleShowPassword = event => {
        this.setState( prevState => {
            return { showPassword: !prevState.showPassword };
        });
        if (event) event.preventDefault();
    };

    render() {

        const {name, value, label, onChange, placeholder, ...props } = this.props;

        return (
            <div style={this.css.for.container}>
                <TextInput
                    id={name}
                    name={name}
                    label={label}
                    placeholder={placeholder}
                    value={value}
                    required={true}
                    type={"password"}
                    onChange={onChange}
                    {...props}>
                </TextInput>
            </div>
        );
    }
}

PasswordInput.propTypes = {
    /** Input name. Recommend setting this to match object's property so a single change handler can */
    name: PropTypes.string.isRequired,

    /** Input label  */
    label: PropTypes.string.isRequired,

    /** Function to call onChange */
    onChange: PropTypes.func.isRequired,

    /** Placeholder to display when empty */
    placeholder: PropTypes.string,

    /** Maximum size of password field */
    maxLength: PropTypes.number,

    /** show visibility */
    showVisibilityToggle: PropTypes.bool,

    /** Quality  */
    quality: PropTypes.number,

    /** Value  */
    value: PropTypes.any,

    /** String to display when error occurs.  */
    error: PropTypes.string,
};

PasswordInput.defaultProps = {
    maxLength: 30,
    label: "Password"
};

export default PasswordInput;