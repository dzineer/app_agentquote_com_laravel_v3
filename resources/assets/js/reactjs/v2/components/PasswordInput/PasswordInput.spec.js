import React from 'react';
import renderer from 'react-test-renderer';
import '../../SetupTests';
import PasswordInput from './PasswordInput';
import { shallow } from 'enzyme';

test('toggles input type when show/hide password clicked', () => {
    const wrapper = shallow(
        <PasswordInput
            htmlId="test"
            name="test"
            onChange={() => {}}
            showVisibilityToggle
            value="Uisi38#8iad"
        />
    );
    // Password input should have a type of password initially
    expect(wrapper.find({ type: 'password' })).toHaveLength(1);
    expect(wrapper.find({ type: 'text' })).toHaveLength(0);


    wrapper.find('a').simulate('click');

    // Password input should have a type of test after clicking toggle
    expect(wrapper.find({ type: 'password' })).toHaveLength(0);
    expect(wrapper.find({ type: 'text' })).toHaveLength(1);

});

test('hides password quality by default', () => {
    const tree = renderer.create(
        <PasswordInput
            htmlId="test"
            name="test"
            onChange={() => {}}
            value="Uisi38#8iad"
        />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('shows password quality when enabled and a password is entered', () => {
    const tree = renderer.create(
        <PasswordInput
            htmlId="test"
            name="test"
            onChange={() => {}}
            showQuality
            value="Uisi38#8iad"
        />).toJSON();
    expect(tree).toMatchSnapshot();
});