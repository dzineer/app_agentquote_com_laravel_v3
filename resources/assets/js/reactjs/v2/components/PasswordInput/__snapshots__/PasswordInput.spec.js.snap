// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides password quality by default 1`] = `
<div
  style={
    Object {
      "marginBottom": "16px",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "16px",
      }
    }
  >
    <label
      htmlFor="test"
      style={
        Object {
          "display": "block",
        }
      }
    >
      Password
       
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
         *
      </span>
    </label>
    <input
      id="test"
      maxLength={30}
      name="test"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      style={undefined}
      type="password"
      value="Uisi38#8iad"
    />
  </div>
</div>
`;

exports[`shows password quality when enabled and a password is entered 1`] = `
<div
  style={
    Object {
      "marginBottom": "16px",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "16px",
      }
    }
  >
    <label
      htmlFor="test"
      style={
        Object {
          "display": "block",
        }
      }
    >
      Password
       
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
         *
      </span>
    </label>
    <input
      id="test"
      maxLength={30}
      name="test"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      showQuality={true}
      style={undefined}
      type="password"
      value="Uisi38#8iad"
    />
  </div>
</div>
`;
